swagger.yaml

```diff
@@ -0,0 +1,512 @@
+      responses:
+        '200':
+          description: OK
+  /v1/group:
+    post:
+      tags:
+        - HTTP API
+      summary: Group
+      description: >-
+        The group call lets you link an identified user with a group such as a
+        company, organization, or an account. It also lets you record any custom
+        traits associated with that group, like the name of the company, the
+        number of employees, etc.
+      operationId: Group
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: body
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/GroupPayload'
+      responses:
+        '200':
+          description: OK
+  /v1/alias:
+    post:
+      tags:
+        - HTTP API
+      summary: Alias
+      description: The alias call lets you merge different identities of a known user.
+      operationId: Alias
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: body
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/AliasPayload'
+      responses:
+        '200':
+          description: OK
+  /v1/batch:
+    post:
+      tags:
+        - HTTP API
+      summary: Batch
+      description: >-
+        The batch call enables you to send a batch of events (identify, track,
+        page, group, screen, alias) in a single request.
+      operationId: Batch
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: body
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/BatchPayload'
+      responses:
+        '200':
+          description: OK
+
+  /internal/v1/extract:
+    post:
+      tags:
+        - Internal API
+      summary: Extract
+      description: Handles extract requests.
+      operationId: Extract
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters: []
+      responses:
+        '200':
+          description: OK
+      security:
+        - writeKeyAuth: []
+  /internal/v1/retl:
+    post:
+      tags:
+        - Internal API
+      summary: Retl
+      description: Handles retl (real-time ETL) requests.
+      operationId: Retl
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters: []
+      responses:
+        '200':
+          description: OK
+      security:
+        - writeKeyAuth: []
+  /internal/v1/audiencelist:
+    post:
+      tags:
+        - Internal API
+      summary: Audience List
+      description: Handles audience list requests.
+      operationId: AudienceList
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters: []
+      responses:
+        '200':
+          description: OK
```

Comment:
We might need to add other possible response codes as well, Few samples:
[429 -> Too many requests.](https://github.com/rudderlabs/rudder-server/blob/9cb9caff24ba945825610ca280609c754ed400f2/gateway/handle.go#L408)
[400 -> Invalid JSON](https://github.com/rudderlabs/rudder-server/blob/9cb9caff24ba945825610ca280609c754ed400f2/gateway/handle.go#L292C2-L292C2)
[400 -> Not a rudder event](https://github.com/rudderlabs/rudder-server/blob/9cb9caff24ba945825610ca280609c754ed400f2/gateway/handle.go#L300)
[400 -> Non identifiable](https://github.com/rudderlabs/rudder-server/blob/9cb9caff24ba945825610ca280609c754ed400f2/gateway/handle.go#L340)
[413 -> Request body too large.](https://github.com/rudderlabs/rudder-server/blob/master/gateway/handle.go#L426)